pkg/operator/ceph/cluster/osd/labels.go

```diff
@@ -61,6 +61,12 @@ func (c *Cluster) getOSDLabels(osd OSDInfo, failureDomainValue string, portable
 	labels[deviceClass] = osd.DeviceClass
 	labels[osdStore] = osd.Store
 
+	encryptedOSD := "false"
+	if osd.Encrypted {
```

Comment:
If the encryption setting changes in the CR, does the label still remain accurate on the OSD? We would want to maintain the existing labels until the OSD is replaced.

Reply:
If the encryption setting is changed in the cephCluster CR to `true` or `false`, the OSD deployment will be deleted for migration. The new OSD deployment will have the updated encryption setting based on what was updated in the CephCluster CR.

Reply:
I know this will happen for each single OSD at a time that is being migrated. What I'm not clear about is if all other existing OSDs will be reconciled before they are migrated, and maintain their previous setting. Or will they not be reconciled until they are migrated? Now I can't remember that design point.

Reply:
If migration is pending on existing OSDs, we don't update them.  But already migrated OSDs can update while other OSDs are still migrating.

Reply:
Sounds good, thanks for the clarification